appengine/genetics/js/mouseAvatar.js

```diff
@@ -0,0 +1,307 @@
+  var y3 = radius * 1.5 + yOffset;
+  var centerX = radius + xOffset;
+  var centerY = radius + yOffset;
+
+  // Draw top right slice.
+  var proposeMateSlice = document.createElementNS(Blockly.SVG_NS, 'path');
+  proposeMateSlice.setAttribute('d', 'M ' + x1 + ' ' + y1 +
+      ' A ' + radius + ' ' + radius + ', 0, 0, 1, ' +
+      x2 + ' ' + y2 + ' L ' + centerX + ' ' + centerY + ' Z');
+  proposeMateSlice.setAttribute('fill',
+      Genetics.Visualization.COLOURS[this.proposeMateOwner]);
+  this.element.appendChild(proposeMateSlice);
+
+  // Draw bottom slice.
+  var pickFightSlice = document.createElementNS(Blockly.SVG_NS, 'path');
+  pickFightSlice.setAttribute('d', 'M ' + x2 + ' ' + y2 +
+      ' A ' + radius + ' ' + radius + ', 0, 0, 1, ' +
+      x3 + ' ' + y3 + ' L ' + centerX + ' ' + centerY + ' Z');
+  pickFightSlice.setAttribute('fill',
+      Genetics.Visualization.COLOURS[this.pickFightOwner]);
+  this.element.appendChild(pickFightSlice);
+
+  // Draw top left slice.
+  var acceptMateSlice = document.createElementNS(Blockly.SVG_NS, 'path');
+  acceptMateSlice.setAttribute('d', 'M ' + x3 + ' ' + y3 +
+      ' A ' + radius + ' ' + radius + ', 0, 0, 1, ' +
+      x1 + ' ' + y1 + ' L ' + centerX + ' ' + centerY + ' Z');
+  acceptMateSlice.setAttribute('fill',
+      Genetics.Visualization.COLOURS[this.acceptMateOwner]);
+  this.element.appendChild(acceptMateSlice);
+
+  Object.defineProperty(this, 'direction', {
+    /**
+     * Sets the direction to a value between 0-2PI and rotates the mouse to
+     * match the direction facing. Direction angle is clockwise, with 0
+     * indicating the mouse facing right.
+     * @param {number} direction
+     */
+    set: function(direction) {
+      // Convert direction to a value between 0-2PI
+      while (direction < 0) {
+        direction += 2 * Math.PI;
+      }
+      while (direction > 2 * Math.PI) {
+        direction -= 2 * Math.PI;
+      }
+      // Determine what the change of direction is.
+      var delta = direction - this.direction;
+      if (delta > Math.PI) {
+        delta -= 2 * Math.PI;
+      } else if (delta < -Math.PI) {
+        delta += 2 * Math.PI;
+      }
+      if (Math.abs(delta) > Math.PI/10) {
+        var resetStraight = function() {
+          image.setAttribute('x', '0px');
+        };
+        if (delta > 0) {
+          // If the mouse turned right.
+          image.setAttribute('x', -Genetics.MouseAvatar.WIDTH + 'px');
+          setTimeout(resetStraight, 150)
+        } else if (delta < 0) {
+          // If the mouse turned left.
+          image.setAttribute('x', -2 * Genetics.MouseAvatar.WIDTH + 'px');
+          setTimeout(resetStraight, 150)
+        }
+      }
+      // Rotate mouse image to match direction facing.
+      this.element.style.transform =
+          'rotate(' + (direction + Math.PI / 2) + 'rad)';
+      this.direction_ = direction;
+    },
+    /**
+     * Returns the direction the mouse is facing.
+     * @returns {number}
```

Comment:
It is just @return not @returns (there is another one of these in visualization.js).
Also, this line would be a great place to describe the output as being in radians.